lib/plugins/aws/invokeLocal/index.js

```diff
@@ -443,113 +422,107 @@ class AwsInvokeLocal {
       'invokeLocal',
       'artifact'
     );
-    return decompress(artifact, destination).then(() => destination);
+    await decompress(artifact, destination);
+    return destination;
   }
 
   getEnvVarsFromOptions() {
-    const envVarsFromOptions = {};
     // Get the env vars from command line options in the form of --env KEY=value
-    [].concat(this.options.env || []).forEach(itm => {
+    return [].concat(this.options.env || []).reduce((envVarsFromOptions, itm) => {
       const splitItm = itm.split('=');
       envVarsFromOptions[splitItm[0]] = splitItm.slice(1, splitItm.length).join('=') || '';
-    });
-    return envVarsFromOptions;
+      return envVarsFromOptions;
+    }, {});
```

Comment:
Let's simply construct this as:

```javascript
const env = {};
for (const envOption of [].concat(this.options.env || [])) {
  ..
}
return env;
```

`reduce` feels controversial here, and is not as readable